### PR TITLE
Jrccties 51 newsletter mail

### PR DESCRIPTION
**Pull request by ovidenov**

### CODE REVIEW

File: sites/default/modules/features/jrccties_core/jrccties_core.module [hunk 1]
```diff
@@ -7,6 +7,271 @@
...
+ * Gets list of groups per user.
+ *
+ * @param int $uid
+ *   The user id.
+ *
+ * @return mixed
+ *   An array of group ids.
+ */
+function _jrccties_core_get_groups_per_user($uid) {
+  static $communities = array('alumni_network', 'enlightenment', 'bioeconomy');
```
Comment (amd-miri) [inline sites/default/modules/features/jrccties_core/jrccties_core.module]: Please make it dynamic

File: sites/default/modules/features/jrccties_core/jrccties_core.module [hunk 2]
```diff
@@ -7,6 +7,271 @@
...
+        $content_per_user[] = $latest_content[$gid];
+      }
+    }
+
+    if (!$content_per_user) {
+      continue;
+    }
+
+    $account = user_load($uid);
+    // $d = l(t('unsubscribe'), 'newsletter', $link_options);.
```
Comment (amd-miri) [inline sites/default/modules/features/jrccties_core/jrccties_core.module]: Pls delete the ine

File: sites/default/modules/features/jrccties_core/jrccties_core.module [hunk 3]
```diff
@@ -7,6 +7,271 @@
...
+    );
+
+    $mail = drupal_mail($module, $key, $to, language_default(), $params, NULL, TRUE);
+    if (!$mail['result']) {
+      $error_msg = 'Failed to send Weekly Newsletter';
+      watchdog('jrccties_core', $error_msg, array(), WATCHDOG_ALERT);
+    }
+    else {
+      $account->field_newsletter_timestamp[LANGUAGE_NONE][0]['value'] = strtotime(JRCCTIES_CORE_NEWSLETTER_TIMESTAMP_INCREASE);
+      // user_save($account);
```
Comment (amd-miri) [inline sites/default/modules/features/jrccties_core/jrccties_core.module]: pls uncomment

File: sites/default/modules/features/jrccties_core/jrccties_core.module [hunk 4]
```diff
@@ -7,6 +7,271 @@
...
+    }
+
+  }
+}
+
+/**
+ * Implements hook_init().
+ */
```
Comment (amd-miri) [inline sites/default/modules/features/jrccties_core/jrccties_core.module]: pls delete hook_init().

File: sites/default/modules/features/jrccties_core/jrccties_core.module [hunk 5]
```diff
@@ -7,6 +7,271 @@
...
+  }
+}
+
+/**
+ * Implements hook_og_membership_presave().
+ */
+function jrccties_core_og_membership_presave(OgMembership $og_membership) {
+  global $user;
+
+  $membership_types = array(
```
Comment (amd-miri) [inline sites/default/modules/features/jrccties_core/jrccties_core.module]: pls make it synamic

File: sites/default/modules/features/jrccties_core/jrccties_core.module [hunk 6]
```diff
@@ -511,9 +776,76 @@ function jrccties_core_menu() {
...
+  );
   return $items;
 }
 
+/**
+ * Implements hook_form().
+ */
+function jrccties_core_newsletter_subscription_form($form, &$form_state) {
+  global $user;
+  static $newsletter_communities = array(
```
Comment (amd-miri) [inline sites/default/modules/features/jrccties_core/jrccties_core.module]: pls make it dynamic

File: sites/default/modules/features/jrccties_core/jrccties_core.module [hunk 7]
```diff
@@ -7,6 +7,271 @@
 
 define('JRCCTIES_CORE_OG_ENTITYFORM_AUDIENCE_FIELD', 'og_group_ref');
 
+/**
+ * JRCCTIES_CORE_NEWSLETTER_PERIOD constant.
+ */
+define('JRCCTIES_CORE_NEWSLETTER_PERIOD', '-1 month');
```
Comment (amd-miri) [inline sites/default/modules/features/jrccties_core/jrccties_core.module]: make it a setting pls

File: sites/default/modules/features/jrccties_core/jrccties_core.module [hunk 8]
```diff
@@ -7,6 +7,271 @@
...
+    $account->field_newsletter_timestamp[LANGUAGE_NONE][0]['value'] = $og_membership->created;
+    user_save($account);
+  }
+}
+
+/**
+ * Implements hook_cron().
+ */
+function jrccties_core_cron() {
+  _jrccties_core_generate_weekly_mails();
```
Comment (amd-miri) [inline sites/default/modules/features/jrccties_core/jrccties_core.module]: rename the function

File: sites/default/modules/features/jrccties_core/jrccties_core.module [hunk 9]
```diff
@@ -511,9 +776,76 @@ function jrccties_core_menu() {
     'access arguments' => array('administer site configuration'),
     'type' => MENU_CALLBACK,
   );
+
+  $items['newsletter'] = array(
+    'title' => 'Newsletter subscription',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('jrccties_core_newsletter_subscription_form'),
+    'access arguments' => array('administer site configuration'),
```
Comment (amd-miri) [inline sites/default/modules/features/jrccties_core/jrccties_core.module]: change the permission to edit user profile

File: sites/default/modules/features/jrccties_core/jrccties_core.module (hunk 1 above)
Comment (ovidenov) [reply sites/default/modules/features/jrccties_core/jrccties_core.module]: fixed

File: sites/default/modules/features/jrccties_core/jrccties_core.module (hunk 2 above)
Comment (ovidenov) [reply sites/default/modules/features/jrccties_core/jrccties_core.module]: fixed

File: sites/default/modules/features/jrccties_core/jrccties_core.module (hunk 3 above)
Comment (ovidenov) [reply sites/default/modules/features/jrccties_core/jrccties_core.module]: fixed

File: sites/default/modules/features/jrccties_core/jrccties_core.module (hunk 4 above)
Comment (ovidenov) [reply sites/default/modules/features/jrccties_core/jrccties_core.module]: fixed

File: sites/default/modules/features/jrccties_core/jrccties_core.module (hunk 5 above)
Comment (ovidenov) [reply sites/default/modules/features/jrccties_core/jrccties_core.module]: fixed

File: sites/default/modules/features/jrccties_core/jrccties_core.module (hunk 6 above)
Comment (ovidenov) [reply sites/default/modules/features/jrccties_core/jrccties_core.module]: fixed

File: sites/default/modules/features/jrccties_core/jrccties_core.module (hunk 8 above)
Comment (ovidenov) [reply sites/default/modules/features/jrccties_core/jrccties_core.module]: fixed

File: sites/default/modules/features/jrccties_core/jrccties_core.module (hunk 9 above)
Comment (ovidenov) [reply sites/default/modules/features/jrccties_core/jrccties_core.module]: Fixed

**Pull request merged**